src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp

```diff
@@ -16,18 +18,48 @@ namespace DB
  */
 bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 {
-    if (pos->type != TokenType::Dot)
+    // There's a special case, that a path member can begin with number
+    if (pos->type != TokenType::Dot && pos->type != TokenType::Number)
         return false;
+    if (pos->type != TokenType::Number)
+        ++pos;
```

Comment:
did not understand why do we need this condition `if (pos->type != TokenType::Number)` and same above if we expect number after "." and here we are supposed to skip "." and only after comes a number, from your own example in test: `'$.1key'`

Reply:
".1123" is parsed as a number

Reply:
".123" is parsed as a number, not a dot and a number

Reply:
I'm not sure whether change the lexer will bring some compatibility issues. So make the change here will be safty.

Reply:
please put all this into comment in the code